web/api/v1/api.go

```diff
@@ -148,9 +149,16 @@ func (api *API) Register(r *route.Router) {
 	r.Get("/alertmanagers", instr("alertmanagers", api.alertmanagers))
 }
 
+type queryStats struct {
```

Comment:
Maybe rather than defining this struct here it would be better to organize all possible timers as members of a struct (maybe as pointers so some counters can be `nil` depending on kind of query) in query_stats.go instead of a list of consts (https://github.com/prometheus/prometheus/blob/master/util/stats/query_stats.go#L20-L28). And then return the entire struct (JSON-ized), not just some of the timers. This'd require some refactoring of the timer code.